addons/point_of_sale/static/src/js/models.js

```diff
@@ -1525,14 +1525,18 @@ exports.Orderline = Backbone.Model.extend({
             return false;
         }else if(this.get_discount() > 0){             // we don't merge discounted orderlines
             return false;
-        }else if(price !== orderline.get_product().get_price(orderline.order.pricelist, this.get_quantity())){
+        }else if(!utils.float_is_zero(price - orderline.get_product().get_price(orderline.order.pricelist, this.get_quantity()),
```

Comment:
I think you can retrieve the number of digits from the currency more easily, see for example:
https://github.com/odoo/odoo/blob/11.0/addons/account/static/src/js/reconciliation/reconciliation_model.js#L682

Reply:
:sweat: pardon my stupidity, it was indeed there, right in front of me in the debugger, and still I recalculate it... :cold_sweat: